app/scripts/directives/addSecretToApplication.js

```diff
@@ -84,7 +84,17 @@
     ctrl.$onInit = function() {
       ctrl.addType = 'env';
       ctrl.disableInputs = false;
+      ctrl.invalidEnvVars = [];
+
       getApplications();
+
+      var keyValidator = new RegExp("[A-Za-z_][A-Za-z0-9_]*");
+
+      _.each(_.keys(ctrl.secret.data), function(nextKey) {
+        if (!keyValidator.test(nextKey)) {
+          ctrl.invalidEnvVars.push(nextKey);
+        }
+      });
```

Comment:
If you only care if some key is invalid, this could be

```js
ctrl.hasInvalidEnvVars = _.some(ctrl.secret.data, function(value, key) {
  return !keyValidator.test(key);
});
```

Otherwise, I'd probably avoid the extra `keys` call by using

```js
_.each(ctrl.secret.data, function(value, key) {
  if (!keyValidator.test(key)) {
    ctrl.invalidEnvVars.push(key);
  }
});
```

Reply:
Yeah, I was going do do that but wasn't sure if we wanted to be able to list the invalid keys at some point. Thoughts?

Reply:
Still working on getting the validator to work correctly. 
      var keyValidator = new RegExp("[A-Za-z_][A-Za-z0-9_]*");

is allowing 'database-name' (as well as 'database=name')...

Reply:
Maybe

```js
var keyValidator = new RegExp("^[A-Za-z_][A-Za-z0-9_]+$");
```

Reply:
I believe I've got it working with:
```javascript
      var keyValidator = new RegExp("^[A-Za-z_]{1}[A-Za-z0-9_]{0,}$");
```

Reply:
Isn't `*` OK instead of `{0,}`?

Reply:
Yes, it is. Got carried away with some other attempts... ;)